spring-cloud-stream/src/main/java/org/springframework/cloud/stream/binding/BinderAwareChannelResolver.java

```diff
@@ -35,10 +40,12 @@
  * resolves the channel from the bean factory and, if not present, creates a new channel
  * and adds it to the factory after binding it to the binder. The binder is optionally
  * determined with a prefix preceding a colon.
+ *
  * @author Mark Fisher
  * @author Gary Russell
+ * @author Ilayaperumal Gopinathan
  */
-public class BinderAwareChannelResolver extends BeanFactoryMessageChannelDestinationResolver {
+public class BinderAwareChannelResolver extends BeanFactoryMessageChannelDestinationResolver implements SmartLifecycle {
```

Comment:
Maybe I am missing something, but `BinderAwareChannelResolver` is created  by `ChannelBindingServiceConfiguration`, and only depends on the `BinderFactory` and the `ChannelBindingServiceProperties` whereas  `Bindable`s are referenced in `ChannelBindingAutoConfiguration`, which happens later. We already have a `Bindable` created in `ChannelBindingServiceConfiguration`. Not sure where the circular dependency is coming from.